src/Compilers/CSharp/Portable/Syntax/DirectiveTriviaSyntax.cs

```diff
@@ -72,20 +72,20 @@ public SyntaxToken DirectiveNameToken
             {
                 foreach (var tr in token.LeadingTrivia)
                 {
-                    if (next)
+                    if (tr.IsDirective)
                     {
-                        if (tr.IsDirective)
+                        var d = (DirectiveTriviaSyntax)tr.GetStructure()!;
```

Comment:
>(DirectiveTriviaSyntax)tr.GetStructure() [](http://example.com/codeflow?start=32&length=40)

We're now calling `SyntaxToken.GetStructure()` for all directives before the target directive. Is that expensive? Same question for `GetPreviousDirective()`.

Reply:
@cston I'd imagine that'd only be happening if you have multiple directives in the same block of trivia, which would be a very tiny amount of extra work. I'd aim for simpler code here.

Reply:
> that'd only be happening if you have multiple directives in the same block of trivia

Thanks, I'd missed that.